File: Entities/Models/AffiliateUser.cs

using Entities.Enums;

namespace Entities.Models
{
    public class AffiliateUser
    {
        public int Id { get; set; }
        public string ReferralCode { get; set; }

        public int ReferralCount { get; set; } = 0;
        public double TotalEarnings { get; set; } = 0.0;
        public double CommissionRate { get; set; } = 0.0;


        public double PaidCash { get; set; } = 0.0;
        public double ApprovedCash { get; set; } = 0.0;
        public double PendingCash { get; set; } = 0.0;
        public int Points { get; set; } = 0;

        public string PaymentEmail { get; set; }
        public string PaymentGateway { get; set; }

        public string UserId { get; set; }
        public string VisitorID { get; set; }

    }
}

File: eCommerce/ClientApp/src/app/shared/classes/affiliateUser.ts
export enum CommissionStatus {
  Pending = 0,
  Approved = 1,
  Paid = 2
}

export interface AffiliateUser {
  id?: number;
  userId?: string;
  paymentEmail: string;
  paymentGateway: string;
  referralCode?: string;

  referralCount?: number;
  totalEarnings?: number;
  commissionRate?: number;
  status?: CommissionStatus;

  points?: number;
  paidCash?: number;
  approvedCash?: number;
  pendingCash?: number;
}

export interface PerformanceChart {
  date: string;
  totalRevenue: number;
  orderCount: number;
}

export interface PaymentDetails {
  userId: string;
  paymentEmail: string;
  paymentGateway: string;
}
